### ISSUE TITLE
Modified MacIcon.icns works in Linux but not in Mac?

### ISSUE THREAD

**Issue by LaraSQP** [picoe/Eto#1501]
I have modified the `MacIcon.icns` and compiled a cross-platform project.

`\Test.Desktop.app\Contents\MonoBundle\MacIcon.icns` contains the modified `MacIcon.icns` file.

`\Test.Desktop.app\Contents\Resources\MacIcon.icns` contains the unmodified default `MacIcon.icns` file.

I can replace the unmodifed file manually or via post-build but, what gives?

As before, I am new to eto so the error could be mine.

Thank you all.

**Comment by cwensley** [comment]
Hey @LaraSQP I'm not sure how the icns is working on linux, as icns files are _only_ for macOS.

By default, for Eto 2.5 you simply need to supply an Icon.icns file in the top level of your project, assuming you are using the Eto.Platforms.Mac64 NuGet package.

**Comment by LaraSQP** [comment]
I have just created a project and all I did was replace the `icns` file with my own.

After building the release version, I have the following folder structure:

```
MacOS\
MonoBundle\
Resources\
```
Inside the `Resources` is the unmodified (original) icns file. I have no idea where it is being fetched from.

Having said that, only the contents of the folder `MacOS` are needed for the Mac port, right?

Or does the Mac port need the other folders too?

(The `MonoBundle` folder does contain the modified `icns` file)

Apologies if this is a stupid question.

**Comment by cwensley** [comment]
Are you using Eto 2.5?

**Comment by LaraSQP** [comment]
Darn, I am not. Let me try again.

**Comment by LaraSQP** [comment]
So, I updated to `2.5.0-rc.4`, deleted obj/bin folders, did a clean rebuild and now have following folder structure:

```
MacOS\
Resources\
```

As before, inside the Resources is the unmodified (original) `icns` file. I have no idea where it is being fetched from.

Note that this is a new project with no other modification than the icns file.

**Comment by cwensley** [comment]
In Eto 2.5, just create an 'Icon.icns' file in the root of your project, then set its build action to "BundleResource". it should pick it up from there.

If you have an Info.plist file with a different .icns file specified for the `CFBundleIconFile` key then it'll use that instead.  See the [Eto.Test.Mac64](https://github.com/picoe/Eto/tree/develop/test/Eto.Test.Mac) app on how that is done.

It is not the same structure as 2.4, and will probably need some additional documentation.

**Comment by LaraSQP** [comment]
Do not have a `BundleResource` option but have an `Embedded resource`.

It did not work, though.

**Comment by cwensley** [comment]
Try Content.  The option might only be visible using VS on Mac.

**Comment by LaraSQP** [comment]
That's the one.

`Content` did it.

Your patience is much appreciated.
